flagd/pkg/service/flag-sync/sync_service.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"fmt"
+	"net"
+
+	"buf.build/gen/go/open-feature/flagd/grpc/go/flagd/sync/v1/syncv1grpc"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/store"
+	"google.golang.org/grpc"
+)
+
+type ISyncService interface {
+	// Start the sync service
+	Start() error
+	// Emit updates for sync listeners
+	Emit()
+	// Shutdown the sync service
+	Shutdown()
+}
+
+type SvcConfigurations struct {
+	Logger  *logger.Logger
+	Port    uint16
+	Sources []string
+	Store   *store.Flags
+}
+
+type Service struct {
+	listener net.Listener
+	logger   *logger.Logger
+	mux      *Multiplexer
+	server   *grpc.Server
+}
+
+func NewSyncService(cfg SvcConfigurations) (*Service, error) {
+	l := cfg.Logger
+	mux, err := NewMux(cfg.Store, cfg.Sources)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing multiplexer: %w", err)
+	}
+
+	server := grpc.NewServer()
+	syncv1grpc.RegisterFlagSyncServiceServer(server, &syncHandler{
+		mux: mux,
+		log: l,
+	})
+
+	l.Info(fmt.Sprintf("starting flag sync service on port %d", cfg.Port))
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.Port))
+	if err != nil {
+		return nil, fmt.Errorf("error creating listener: %w", err)
+	}
+
+	return &Service{
+		listener: listener,
+		logger:   l,
+		mux:      mux,
+		server:   server,
+	}, nil
+}
+
+func (s *Service) Start() error {
+	err := s.server.Serve(s.listener)
+	if err != nil {
+		return fmt.Errorf("error from server start: %w", err)
+	}
+
+	return nil
+}
```

Comment:
I dunno if it's something we should add here, but I think there's some missing context/cancellation handling somewhere.

If I send a `SIGINT` to flagd WITHOUT the `-e` flag, it shuts down fine within a couple seconds. If I send a `SIGINT` WITH the `-e` flag, flagd never shuts down.

Reply:
Good observation. I checked the shutdown logic and improved it to respect the SIGTERM. Updated the PR with commit https://github.com/open-feature/flagd/pull/1237/commits/711a21b916f8f4e65aa8c30fcc58b8df8dcc3293

Reply:
Tested and works great now.